eng/cake/dotnet.cake

```diff
@@ -0,0 +1,195 @@
+// Contains .NET 6-related Cake targets
+
+var ext = IsRunningOnWindows() ? ".exe" : "";
+var dotnetPath = $"./bin/dotnet/dotnet{ext}";
+
+Task("dotnet")
+    .Description("Provisions .NET 6 into bin/dotnet based on eng/Versions.props")
+    .Does(() =>
+    {
+        var binlog = $"artifacts/dotnet-{configuration}.binlog";
+        var settings = new DotNetCoreBuildSettings
+        {
+            MSBuildSettings = new DotNetCoreMSBuildSettings()
+                .EnableBinaryLogger(binlog)
+                .SetConfiguration(configuration),
+        };
+        DotNetCoreBuild("./src/DotNet/DotNet.csproj");
```

Comment:
This `.binlog` didn't show up. I think because `settings` didn't get passed in, my bad:
```suggestion
        DotNetCoreBuild("./src/DotNet/DotNet.csproj", settings);
```